src/app/profiles/profile-detail/profile-detail.component.ts

```diff
@@ -356,11 +359,11 @@ export class ProfileDetailComponent implements OnInit {
       const result: any = Object.assign({}, this.profileForm.getRawValue())
       // When creating new
       if ((result.connectionMode === 'CIRA' && result.dhcpEnabled === false) &&
-      (!this.isEdit && (result.generateRandomPassword || result.generateRandomMEBxPassword))) {
+      (!this.isEdit && ((result.activation === Constants.ACMActivate && (result.generateRandomPassword || result.generateRandomMEBxPassword)) || (result.activation === Constants.CCMActivate && result.generateRandomPassword)))) {
```

Comment:
I think this works, however, this whole function has grown to be a bit hard to follow. You wanna take a stab at refactoring a bit? First thing I would look at it reducing the calls to the various dialog functions? And try to reduce the places we call `.onSubmit()?`